src/api/forms/service.js

```diff
@@ -163,12 +164,39 @@ export async function createForm(metadataInput, author) {
 }
 
 /**
- * Lists the available form metadata
+ * Lists forms and returns query result metadata (e.g., pagination details, total counts)
+ * @param {PaginationOptions} [options] - Optional pagination options
+ * @returns {Promise<FormMetadata[] | QueryResult<FormMetadata>>}
  */
-export async function listForms() {
-  const documents = await formMetadata.list()
+export async function listForms(options) {
+  let documents
 
-  return documents.map(mapForm)
+  if (!options?.page && !options?.perPage) {
```

Comment:
How about we change paginationOptionsSchema to provide defaults which allows us to remove the old `listAll` completely?

```
Joi.object().keys({
          page: Joi.number().positive().integer().min(1).default(1),
          perPage: Joi.number()
            .positive()
            .integer()
            .min(1)
            .max(200)
            .default(100)
        })
``` 

This would allow the endpoint to be consistent in what it returns (@returns {~~Promise<FormMetadata[] |~~ QueryResult<FormMetadata>>})

Reply:
The concern here is always returning an array of forms, if no pagination is applied so the consuming client(s) don't break once this is merged. That being said, I think this is definitely something we could have in a clean-up ticket once the client(s) are updated to parse the paginated response?

Reply:
There's only one consumer of this; the `/library` page in designer. I thought we'd be updating that (to look for `result.data`) alongside this change. If not, then yeah let's do it after this lands.

Reply:
That ticket was created here: https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/483335